PICS/Schedule/EventContext.swift

```diff
@@ -8,18 +8,39 @@
 
 import SpeziScheduler
 
+// struct EventContext: Comparable, Identifiable {
+//    let event: Event
+//    let task: Task<PICSTaskContext>
+//    
+//    
+//    var id: Event.ID {
+//        event.id
+//    }
+//    
+//    
+//    static func < (lhs: EventContext, rhs: EventContext) -> Bool {
+//        lhs.event.scheduledAt < rhs.event.scheduledAt
+//    }
+// }
 
-struct EventContext: Comparable, Identifiable {
+struct EventContext: Comparable, Identifiable, Hashable {
     let event: Event
     let task: Task<PICSTaskContext>
     
-    
     var id: Event.ID {
         event.id
     }
     
-    
     static func < (lhs: EventContext, rhs: EventContext) -> Bool {
         lhs.event.scheduledAt < rhs.event.scheduledAt
     }
+
+    static func == (lhs: EventContext, rhs: EventContext) -> Bool {
+        lhs.event == rhs.event && lhs.task == rhs.task
+    }
+
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(event)
+        hasher.combine(task)
+    }
```

Comment:
It seems there is a warning that these functions are not covered by tests but not sure how important that it.